exercise5-2.go

```diff
@@ -33,16 +38,18 @@ func lissajous(out io.Writer) {
 	phase := 0.0 // phase difference
 	for i := 0; i < nframes; i++ {
 		rect := image.Rect(0, 0, 2*size+1, 2*size+1)
-		img := image.NewPaletted(rect, palette)
+		img := image.NewPaletted(rect, palette2)
+
 		for t := 0.0; t < cycles*2*math.Pi; t += res {
 			x := math.Sin(t)
 			y := math.Sin(t*freq + phase)
-			img.SetColorIndex(size+int(x*size+0.5), size+int(y*size+0.5),
-				blackIndex)
+			rand.Seed(time.Now().Unix())
```

Comment:
`rand` should only need to be seeded once.  You're re-seeding it inside of a tight loop, which will result in less randomness -- e.g. if the loop iterates several times before the next time `time.Now().Unix()` ticks up, this will result in the exact same "random" value being generated because the state is reset each time. Try the following:

```go
package main

import (
        "fmt"
        "math/rand"
        "time"
)

func main() {
        fmt.Println("Seeded in loop:")
        for i := 0; i < 10; i += 1 {
                rand.Seed(7)
                fmt.Println(rand.Int())
        }
        fmt.Println("Seeded in loop with time:")
        for i := 0; i < 10; i += 1 {
                rand.Seed(time.Now().Unix())
                fmt.Println(rand.Int())
        }
        fmt.Println("Seeded outside of loop:")
        rand.Seed(7)
        for i := 0; i < 10; i += 1 {
                fmt.Println(rand.Int())
        }
}
```